ste/sourceInfoProvider-Local_linux.go

```diff
@@ -0,0 +1,189 @@
+// +build linux
+
+package ste
+
+import (
+	"errors"
+	"github.com/Azure/azure-storage-azcopy/v10/common"
+	"golang.org/x/sys/unix"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func (f localFileSourceInfoProvider) HasUNIXProperties() bool {
+	return true
+}
+
+func (f localFileSourceInfoProvider) GetUNIXProperties() (common.UnixStatAdapter, error) {
+	// Can we use statx?
+	var uname unix.Utsname
+	err := unix.Uname(&uname)
+	if err != nil {
+		return nil, err
+	}
+
+	linuxVersion := regexp.MustCompile("(\\d+\\.\\d+\\.\\d+)").FindString(string(uname.Release[:]))
+	splits := strings.Split(linuxVersion, ".")
+
+	if len(splits) != 3 {
+		return nil, errors.New("Failed to parse linux version " + linuxVersion)
+	}
+
+	versions := make([]int64, 3)
+	for k, v := range splits {
+		versions[k], _ = strconv.ParseInt(v, 10, 64)
+	}
+
+	var resp common.UnixStatAdapter
+
+	if versions[0] > 4 || (versions[0] == 4 && versions[1] > 11) { // Can we statx?
+		var stat unix.Statx_t
+		// dirfd is a null pointer, because we should only ever be passing relative paths here, and directories will be passed via transferInfo.Source.
+		// AT_SYMLINK_NOFOLLOW is not used, because we automagically resolve symlinks. TODO: Add option to not follow symlinks, and use AT_SYMLINK_NOFOLLOW when resolving is disabled.
```

Comment:
I think for our usecase (backing up files) AT_SYMLINK_NOFOLLOW should be default. This is because when looking at a symlink file property we will be interested in the properties of the symlink and not of the target. If we need target properties we can always evaluate the symlink and then call statx() on the target.

Reply:
Any reason why you are not using AT_SYMLINK_NOFOLLOW?
As I said, GetUNIXProperties is a low level call for getting file attributes. It should truthfully get the attributes of the requested file. Whether to follow a symlink or not should be a higher level decision (or if u want pass an option to GetUNIXProperties).
If we don't use AT_SYMLINK_NOFOLLOW  there's no way for the caller to get the symlink properties for cases where it decides to not expand symlinks and copy them as is.

Reply:
I think once we shift to uploading the symlink path, we should opt for AT_SYMLINK_NOFOLLOW. For now, we still upload the file on the other end of the symlink, so it makes sense to upload the properties of the file on the other end of the symlink.